packages/editor/src/components/post-last-revision/index.js

```diff
@@ -13,7 +13,16 @@ import { addQueryArgs } from '@wordpress/url';
 import PostLastRevisionCheck from './check';
 import { store as editorStore } from '../../store';
 
-function LastRevision( { lastRevisionId, revisionsCount } ) {
+function LastRevision() {
+	const { lastRevisionId, revisionsCount } = useSelect( ( select ) => {
+		const { getCurrentPostLastRevisionId, getCurrentPostRevisionsCount } =
+			select( editorStore );
+		return {
+			lastRevisionId: getCurrentPostLastRevisionId(),
+			revisionsCount: getCurrentPostRevisionsCount(),
+		};
+	}, [] );
```

Comment:
This is duplicated between this component and the underlying `PostLastRevisionCheck` component, could be a good opportunity to extract to a hook.

Reply:
That's true, and I think there are multiple opportunities like this. There's also the `PageAttributesCheck` component, which duplicates code from the generic `PostTypeSupportCheck`. It could easily be converted into a simple `<PostTypeSupportCheck supportsKeys="page-attributes'>` instance.

But I didn't want to make too many diverse changes at once in this PR.

Reply:
Agreed on not making this PR too large. Definitely material for a follow-up if you want to pursuit it.